theano/sandbox/gpuarray/__init__.py

```diff
@@ -52,11 +56,21 @@ def init_dev(dev):
             optdb.add_tags('gpuarray_opt', 'fast_run', 'fast_compile')
         elif (config.init_gpu_device.startswith('cuda') or
               config.init_gpu_device.startswith('opencl')):
+            if config.device != 'cpu':
```

Comment:
What do you mean? If `init_gpu_device` is specified and `device` is _not_ `cpu`, then it is an error.
The old backend has `assert device == 'cpu'`, which is equivalent.